dnsprovider/google.go

```diff
@@ -79,126 +88,90 @@ func (p *GoogleProvider) DeleteZone(name string) error {
 	return nil
 }
 
-// Records returns the list of records in a given hosted zone.
-func (p *GoogleProvider) Records(zone string) ([]endpoint.Endpoint, error) {
-	records, err := p.ResourceRecordSetsClient.List(p.Project, zone).Do()
-	if err != nil {
-		return nil, err
-	}
-
-	endpoints := []endpoint.Endpoint{}
-
-	for _, r := range records.Rrsets {
-		if r.Type != "A" {
-			continue
+// Records returns the list of A records in a given hosted zone.
+func (p *GoogleProvider) Records(zone string) (endpoints []endpoint.Endpoint, _ error) {
+	f := func(resp *dns.ResourceRecordSetsListResponse) error {
+		for _, r := range resp.Rrsets {
+			if r.Type != "A" {
+				continue
+			}
+
+			for _, rr := range r.Rrdatas {
+				endpoints = append(endpoints, endpoint.Endpoint{
```

Comment:
yup seems right 👍 , maybe just a small comment would be appropriate, that the operation is bug-free because pagination is sequential :)